skyve-web/src/main/java/org/skyve/impl/web/WebUtil.java

```diff
@@ -416,24 +422,60 @@ public static String resetPassword(String passwordResetToken, String newPassword
 	throws Exception {
 		String customerName = null;
 		String userName = null;
+		Timestamp tokenCreated = null;
 		String errorMsg = null;
+
 		try (Connection c = EXT.getDataStoreConnection()) {
-			try (PreparedStatement s = c.prepareStatement(String.format("select %s, %s from ADM_SecurityUser where %s = ?",
-																			Bean.CUSTOMER_NAME,
-																			AppConstants.USER_NAME_ATTRIBUTE_NAME,
-																			AppConstants.PASSWORD_RESET_TOKEN_ATTRIBUTE_NAME))) {
-				s.setString(1, passwordResetToken);
-				try (ResultSet rs = s.executeQuery()) {
-					if (!rs.isBeforeFirst() ) {
-						return "Reset link used is invalid";
+			// Get password reset token expiry minutes
+			try (PreparedStatement s2 = c.prepareStatement(String.format("select %s from ADM_Configuration",
+					AppConstants.PASSWORD_RESET_TOKEN_EXPIRY_MINUTES_ATTRIBUTE_NAME))) {
+				try (ResultSet rs2 = s2.executeQuery()) {
+					Integer expiryMinutes = null;
+					if (rs2.next()) {
+						int value = rs2.getInt(1);
+						if (!rs2.wasNull()) {
+							expiryMinutes = Integer.valueOf(value);
+						}
 					}
-					while (rs.next()) {
-						customerName = rs.getString(1);
-						userName = rs.getString(2);
 
-						Repository r = CORE.getRepository();
-						org.skyve.metadata.user.User u = r.retrieveUser(String.format("%s/%s", customerName, userName));
-						errorMsg = makePasswordChange(u, null, newPassword, confirmPassword);
+					// Get user by token
+					try (PreparedStatement s = c
+							.prepareStatement(String.format("select %s, %s, %s from ADM_SecurityUser where %s = ?",
+									Bean.CUSTOMER_NAME,
+									AppConstants.USER_NAME_ATTRIBUTE_NAME,
+									AppConstants.PASSWORD_RESET_TOKEN_CREATION_DATE_TIME_ATTRIBUTE_NAME,
+									AppConstants.PASSWORD_RESET_TOKEN_ATTRIBUTE_NAME))) {
+						s.setString(1, passwordResetToken);
+
+						try (ResultSet rs = s.executeQuery()) {
+							if (!rs.isBeforeFirst()) {
+								return "Reset link used is invalid";
+							}
+							while (rs.next()) {
+								customerName = rs.getString(1);
+								userName = rs.getString(2);
+								tokenCreated = rs.getTimestamp(3);
+								
+								// Check for expiry of token
+								boolean expired = false;
+								if (expiryMinutes != null) {
+									Timestamp now = Timestamp.from(Instant.now());
+									if (tokenCreated != null) {
+										TimeUtil.addMinutes(tokenCreated, expiryMinutes.intValue());
+										if (now.after(tokenCreated)) {
+											expired = true;
+										}
+									}
+								}
+								if (!expired) {
+									Repository r = CORE.getRepository();
+									org.skyve.metadata.user.User u = r.retrieveUser(String.format("%s/%s", customerName, userName));
+									errorMsg = makePasswordChange(u, null, newPassword, confirmPassword);
+								} else {
+									return "Token has expired - please reset password again";
```

Comment:
Can this go into the `i18n` files? I'm not sure if we can, can you check if there are any other examples in WebUtil and if it has access to the internationalisation stuff in Skyve to externalise this error message?